File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/serializer/BigIntegerToJsonSerializer.java
package org.javamaster.invocationlab.admin.serializer;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.core.JsonToken;
import com.fasterxml.jackson.core.type.WritableTypeId;
import com.fasterxml.jackson.databind.JsonSerializer;
import com.fasterxml.jackson.databind.SerializerProvider;
import com.fasterxml.jackson.databind.jsontype.TypeSerializer;

import java.io.IOException;
import java.math.BigInteger;

/**
 * BigInteger转换成string返回
 *
 * @author yudong
 */
public class BigIntegerToJsonSerializer extends JsonSerializer<BigInteger> {
    public static final BigIntegerToJsonSerializer INSTANCE = new BigIntegerToJsonSerializer();

    @Override
    public void serialize(BigInteger value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
        if (value == null) {
            gen.writeNull();
        } else {
            gen.writeString(value.toString());
        }
    }

    @Override
    public void serializeWithType(BigInteger value, JsonGenerator gen, SerializerProvider serializers, TypeSerializer typeSer)
            throws IOException {
        WritableTypeId typeIdDef = typeSer.writeTypePrefix(gen, typeSer.typeId(value, JsonToken.VALUE_NUMBER_INT));
        serialize(value, gen, serializers);
        typeSer.writeTypeSuffix(gen, typeIdDef);
    }

}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/serializer/BigIntegerToJsonSerializer.kt
package org.javamaster.invocationlab.admin.serializer

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.core.JsonToken
import com.fasterxml.jackson.databind.JsonSerializer
import com.fasterxml.jackson.databind.SerializerProvider
import com.fasterxml.jackson.databind.jsontype.TypeSerializer
import java.math.BigInteger

/**
 * BigInteger转换成string返回
 *
 * @author yudong
 */
class BigIntegerToJsonSerializer : JsonSerializer<BigInteger?>() {

    override fun serialize(value: BigInteger?, gen: JsonGenerator, serializers: SerializerProvider) {
        if (value == null) {
            gen.writeNull()
        } else {
            gen.writeString(value.toString())
        }
    }

    override fun serializeWithType(
        value: BigInteger?,
        gen: JsonGenerator,
        serializers: SerializerProvider,
        typeSer: TypeSerializer
    ) {
        val typeIdDef = typeSer.writeTypePrefix(gen, typeSer.typeId(value, JsonToken.VALUE_NUMBER_INT))
        serialize(value, gen, serializers)
        typeSer.writeTypeSuffix(gen, typeIdDef)
    }

    companion object {
        val INSTANCE: BigIntegerToJsonSerializer = BigIntegerToJsonSerializer()
    }
}
